docs/production/management-commands.md

```diff
@@ -64,6 +64,32 @@ Otherwise, the `string_id` will correspond to the organization's
 subdomain.  E.g. on `it.zulip.example.com`, use
 `/home/zulip/deployments/current/manage.py show_admins -r it`.
 
+### Deleting a user and their data
+
+Whereas by default we recommend simply deactivating user accounts,
+sometimes there is a need to entirely delete the user and their data.
+You can do that using the `./manage.py delete_user` command. This will:
+* Delete the user's account.
+* Delete any messages they've sent and any personal messages they've received.
+  Note that this applies strictly to personal (between two users) messages.
+  Messages in group conversations in which the user participated won't be deleted
+  (with the exceptions of those message the user sent). Instead, an inactive,
+  inaccessible dummy user account is created to replace the deleted user in those
+  historical group conversation, in order to preserve their integrity.
+* Deactivate all bots owned by the user, without deleting them or their data.
+  If you want to delete the bots and the message sent/received by them, you can
+  use the command on them individually.
+
+An example use can look like this:
+```
+$ ./manage.py delete_user -r zulip hamlet@zulip.com
+hamlet@zulip.com has 0 active bots that will be deactivated as a result of the user's deletion.
+This was a dry run. Pass -f to actually delete.
+$ ./manage.py delete_user -r zulip hamlet@zulip.com -f
+hamlet@zulip.com has 0 active bots that will be deactivated as a result of the user's deletion.
+Successfully deleted the user.
+```
```

Comment:
This is kinda long; let's put this in "Other useful manage.py commands" so it doesn't break the general information flow of this page.  We can first document `manage.py deactivate_user` there -- and mention that it's better to do it via the UI.  

And then we can mention this in that context.  Ideally, we don't need the example use output (because the `--help` should be good enough; we can just mention:

```
* `manage.py deleter_user`: Fully delete a user from the server.  See [delete a user](/help/delete-user) for details on
the exact semantics of this operation; in most cases, it is better to deactivate the user.
```

and then we write a /help/ article that for this that explains the background for how it works?  And can mention that you need to email us if you want this on Zulip Cloud.